frontend/src/components/GeyserStakeView.tsx

```diff
@@ -66,6 +67,90 @@ export const GeyserStakeView: React.FC<Props> = () => {
+    refreshInputAmount()
+    refreshWalletAmount()
+  }
+
+  const withdrawStaking = async () => {
+    if (withdrawFromVault)
+      return withdrawFromVault(stakingTokenAddress, parsedUserInput)
+    return undefined
+  }
+
+  const withdrawReward = async (receipt?: TransactionReceipt) => {
+    if (receipt && withdrawRewardsFromVault) {
+      const tx = await withdrawRewardsFromVault(receipt)
+      if (tx) {
+        const { response, rewards } = tx
+        setActualRewardsFromUnstake(rewards)
+        return response
+      }
+    }
+    return undefined
+  }
+
+  const withdrawStakingTxMessage = (txStateMachine: TxStateMachine) => {
+    const { state, response } = txStateMachine
+    switch (state) {
+      case TxState.PENDING:
+        return <span>Withdrawing {stakingTokenSymbol} from your vault...</span>
+      case TxState.SUBMITTED:
+        return (
+          <span>
+            Withdrawing {stakingTokenSymbol} from your vault...{' '}
+            View on <a rel="noreferrer" className="text-link" href={`https://etherscan.io/tx/${response?.hash}`} target="_blank">Etherscan</a>
+          </span>
+        )
+      case TxState.MINED:
+        return (
+          <span>
+            Successfully withdrew <b>{userInput} {stakingTokenSymbol}</b> from your vault to your wallet.{' '}
+            View on <a rel="noreferrer" className="text-link" href={`https://etherscan.io/tx/${response?.hash}`} target="_blank">Etherscan</a>
+          </span>
+        )
+      case TxState.FAILED:
+        return (
+          <span>
+            Unlocked <b>{userInput} {stakingTokenSymbol}</b> that can be withdrawn from your vault.
+          </span>
+        )
+      default:
+        return <></>
+    }
+  }
+
+  const withdrawRewardTxMessage = (txStateMachine: TxStateMachine) => {
+    const { state, response } = txStateMachine
+    switch (state) {
+      case TxState.PENDING:
+        return <span>Withdrawing {rewardTokenSymbol} from your vault...</span>
+      case TxState.SUBMITTED:
+        return (
+          <span>
+            Withdrawing {rewardTokenSymbol} from your vault...{' '}
+            View on <a rel="noreferrer" className="text-link" href={`https://etherscan.io/tx/${response?.hash}`} target="_blank">Etherscan</a>
+          </span>
+        )
+      case TxState.MINED:
+        return (
+          <span>
+            Successfully withdrew <b>{formatUnits(actualRewardsFromUnstake, rewardTokenDecimals)} {rewardTokenSymbol}</b> from your vault to your wallet.{' '}
+            View on <a rel="noreferrer" className="text-link" href={`https://etherscan.io/tx/${response?.hash}`} target="_blank">Etherscan</a>
+          </span>
+        )
+      case TxState.FAILED:
+        return (
+          <span>
+            Unlocked <b>{formatUnits(actualRewardsFromUnstake, rewardTokenDecimals)} {rewardTokenSymbol}</b> that can be withdrawn from your vault.
+          </span>
+        )
+      default:
+        return <></>
+    }
```

Comment:
I think these guys should go in their own util file or a constant

Reply:
Since those return `ReactNode`s, added those in a component `WithdrawTxMessage`